configstack/running_module.go

```diff
@@ -114,10 +114,12 @@ func runModulesWithHandler(modules []*TerraformModule, handler ModuleHandler, or
 	}
 
 	if len(modules) != 0 {
-		if modules[0].TerragruntOptions.NbWorkers <= 0 {
-			modules[0].TerragruntOptions.NbWorkers = len(runningModules)
+		options := modules[0].TerragruntOptions
+		// We don't gain anything by running on more workers than there are modules
+		if options.NbWorkers <= 0 || options.NbWorkers > len(runningModules) {
```

Comment:
I see your point. I don't think it's wrong to simply match the number of workers to the modules because in the end, that's what happens, but with extra steps. Would a log saying something in the neighbourhood of 
> "Hey you set your worker count to 100, but you have need for only 1 per module. Only spawning X workers." 

be good for you? 

That way, it even informs you that you're maybe setting something too high and that it won't do a difference (it wouldn't be farfetched to think one could think bumping the number of workers will boost performance).